utils/obsproc/NetCDFToIodaConverter2D.h

```diff
@@ -0,0 +1,318 @@
+      ASSERT(intMetadataName_ == intMetadataName_);
+
+      // Concatenate Eigen arrays and vectors
+      longitude_.conservativeResize(location_ + other.location_);
+      latitude_.conservativeResize(location_ + other.location_);
+      datetime_.conservativeResize(location_ + other.location_);
+      obsVal_.conservativeResize(location_ * channel_ + other.location_ * other.channel_);
+      obsError_.conservativeResize(location_ * channel_ + other.location_ * other.channel_);
+      preQc_.conservativeResize(location_ * channel_ + other.location_ * other.channel_);
+      floatMetadata_.conservativeResize(location_ + other.location_, nfMetadata_);
+      intMetadata_.conservativeResize(location_ + other.location_, niMetadata_);
+
+      // Copy data from the 'other' object to this object
+      longitude_.tail(other.location_) = other.longitude_;
+      latitude_.tail(other.location_) = other.latitude_;
+      datetime_.tail(other.location_) = other.datetime_;
+      obsVal_.tail(other.location_ * other.channel_) = other.obsVal_;
+      obsError_.tail(other.location_ * other.channel_) = other.obsError_;
+      preQc_.tail(other.location_ * other.channel_) = other.preQc_;
+      floatMetadata_.bottomRows(other.location_) = other.floatMetadata_;
+      intMetadata_.bottomRows(other.location_) = other.intMetadata_;
+
+      // Update obs count
+      location_ += other.location_;
+      oops::Log::trace() << "IodaVars2d::IodaVars2d done appending." << std::endl;
+    }
+  };
+
+  // Base class for the converters
+  class NetCDFToIodaConverter2d {
+   public:
+    // Constructor: Stores the configuration as a data members
+    explicit NetCDFToIodaConverter2d(const eckit::Configuration & fullConfig,
+                                   const eckit::mpi::Comm & comm): comm_(comm),
+                                                                   fullConfig_(fullConfig) {
+      // time window info
+      std::string winbegin;
+      std::string winend;
+      fullConfig.get("window begin", winbegin);
+      fullConfig.get("window end", winend);
+      windowBegin_ = util::DateTime(winbegin);
+      windowEnd_ = util::DateTime(winend);
+      variable_ = "None";
+      oops::Log::info() << "--- Window begin: " << winbegin << std::endl;
+      oops::Log::info() << "--- Window end: " << winend << std::endl;
+
+      // get input netcdf files
+      fullConfig.get("input files", inputFilenames_);
+      oops::Log::info() << "--- Input files: " << inputFilenames_ << std::endl;
+
+      // ioda output file name
+      outputFilename_ = fullConfig.getString("output file");
+      oops::Log::info() << "--- Output files: " << outputFilename_ << std::endl;
+      oops::Log::trace() << "NetCDFToIodaConverter::NetCDFToIodaConverter created." << std::endl;
+    }
+
+    // Method to write out a IODA file (writter called in destructor)
+    void writeToIoda() {
+      // Extract ioda variables from the provider's files
+      int myrank  = comm_.rank();
```

Comment:
we have an issue with mpi, it currently only works with containers, and we haven't implemented anything for the gathering of the metadata. I suggest you remove the mpi implementation.